File: VRP-Core/src/main/java/VRPCore/Models/Vec5.java
package VRPCore.Models;

import org.bukkit.Location;
import org.bukkit.World;

public class Vec5 {
    public Vec5(double _x, double _y, double _z, float _yaw, float _pitch){
        x = _x;
        y = _y;
        z = _z;
        yaw = _yaw;
        pitch = _pitch;
    }

    public Vec5(double _x, double _y, double _z) {
        x = _x;
        y = _y;
        z = _z;
    }

    public Vec5(double all){
        x = all;
        y = all;
        z = all;
        yaw = (float)all;
        pitch = (float)all;
    }

    public Vec5(Location loc) {
        x = loc.getX();
        y = loc.getY();
        z = loc.getZ();
        yaw = loc.getYaw();
        pitch = loc.getPitch();
    }

    public Location toLocation(World world){
        return new Location(world, x, y, z, yaw, pitch);
    }

    public double x;
    public double y;
    public double z;
    public float yaw;
    public float pitch;
}

File: VRP-Core/VRP-Core/src/main/kotlin/Models/Vec5.kt
package VRPCore.Models

import org.bukkit.Location
import org.bukkit.World

class Vec5 {
    constructor(_x: Double, _y: Double, _z: Double, _yaw: Float, _pitch: Float) {
        x = _x
        y = _y
        z = _z
        yaw = _yaw
        pitch = _pitch
    }

    constructor(_x: Double, _y: Double, _z: Double) {
        x = _x
        y = _y
        z = _z
    }

    constructor(all: Double) {
        x = all
        y = all
        z = all
        yaw = all.toFloat()
        pitch = all.toFloat()
    }

    constructor(loc: Location) {
        x = loc.x
        y = loc.y
        z = loc.z
        yaw = loc.yaw
        pitch = loc.pitch
    }

    fun toLocation(world: World?): Location {
        return Location(world, x, y, z, yaw, pitch)
    }

    var x: Double
    var y: Double
    var z: Double
    var yaw = 0f
    var pitch = 0f
}